test/validation/Validator.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2020 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script to execute Discovery Server v2 tests."""
+import logging
+import os
+from pathlib import Path
+
+import validation.shared as shared
+
+import xmltodict
+
+
+def validation_file_path(file_path):
+    """Std file path to validate an snapshot."""
+    return os.path.join(file_path, '~')
+
+
+class Validator(object):
+    """
+    Class to validate an snapshot resulting from a Discovery-Server test.
+
+    Validate the test counting and comparing the number of lines of the
+    output snapshot resulted from the test execution and an a priori well
+    knonw output.
+    """
+
+    def __init__(
+        self,
+        snapshot_file_path,
+        ground_truth_snapshot_file_path,
+        test_params,
+        debug=False,
+        logger=None
+    ):
+        """
+        Build a generic validation object.
+
+        :param snapshot_file_path: The path to the test output.
+        :param disposals: True if it is a disposals test (Default: False).
+        :param server_endpoints: True if there are endpoints in the servers
+            (Default: False).
+        :param logger: The logging object. VALIDATION if None
+            logger is provided.
+        :param debug: True/False to activate/deactivate debug logger.
+        """
+        self.set_logger(logger, debug)
+        self.logger.debug('Creating an instance of {}'.format(type(self)))
+
+        self.snapshot_file_path = self.valid_snapshot_path(snapshot_file_path)
+        self.gt_snapshot_file_path = self.valid_snapshot_path(
+            ground_truth_snapshot_file_path)
+
+        self.val_snapshot = self.parse_xml_snapshot(self.snapshot_file_path)
```

Comment:
This could be done as well inside the `GenenateValidation` and `GroundTurthValidation` because `CountLinesValidation` does not use it.
Also, I would suggest to move it to the `validate_virtual` method, so a possible error can be handled and if the test does not require this validation it will skip the parse

Reply:
But if it cannot be parsed it means that the xml is not properly generated and should trigger an `ERROR` for all validators and not `SKIP` some of them.

Reply:
Finally the function `parse_xml_snapshot()` has been moved to `GroundTruthValidation` and `GenerateValidation` validators so that it will not be executed in `CountLinesValidator` but it will be trigger in error in case the snapshot XML is not valid.